gnomad/assessment/sanity_checks.py

```diff
@@ -190,3 +207,766 @@ def compare_row_counts(ht1: hl.Table, ht2: hl.Table) -> bool:
+                hl.is_defined(t.info[f"AC{delimiter}raw"])
+            ),
+        )
+    )
+    logger.info("Frequency spot check counts: %s", hl.eval(freq_counts))
+
+
+def sample_sum_sanity_checks(
+    t: Union[hl.MatrixTable, hl.Table],
+    sexes: List[str] = SEXES,
+    sample_sum_sets_and_pops: Dict[str, List[str]] = {"": POPS},
+    verbose: bool = False,
+    sort_order: List[str] = SORT_ORDER,
+    delimiter: str = "-",
+    metric_first_label: bool = True,
+) -> None:
+    """
+    Perform sanity checks on sample sums in input Table.
+
+    Computes afresh the sum of annotations for a specified group of annotations, and compare to the annotated version;
+    displays results from checking the sum of the specified annotations in the terminal.
+    Also checks that annotations for all expected sample populations are present.
+
+    :param t: Input Table.
+    :param sexes: List of sexes in table.
+    :param subset_pops: Dict with subset (keys) and list of populations within subset (values). An empty string, e.g. "", should be passed as key with callset pops as value to test entire callset.
+    :param verbose: If True, show top values of annotations being checked, including checks that pass; if False, show only top values of annotations that fail checks.
+    :param metric_first_label: If True, metric precedes label group, e.g. AC-afr-male. If False, label group precedes metric, afr-male-AC.
+    :return: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    field_check_expr = {}
+    field_check_details = {}
+    for subset, pops in sample_sum_sets_and_pops.items():
+        pop_names = pops
+
+        # We do not store the raw callstats for the pop or sex groupings of any subset so only check adj here.
+        field_check_expr_s, field_check_details_s = sample_sum_check(
+            t,
+            subset,
+            dict(group=["adj"], pop=pop_names),
+            sort_order,
+            delimiter,
+            metric_first_label,
+        )
+        field_check_expr.update(field_check_expr_s)
+        field_check_details.update(field_check_details_s)
+        field_check_expr_s, field_check_details_s = sample_sum_check(
+            t,
+            subset,
+            dict(group=["adj"], sex=sexes),
+            sort_order,
+            delimiter,
+            metric_first_label,
+        )
+        field_check_expr.update(field_check_expr_s)
+        field_check_details.update(field_check_details_s)
+        field_check_expr_s, field_check_details_s = sample_sum_check(
+            t,
+            subset,
+            dict(group=["adj"], pop=list(set(pop_names)), sex=sexes),
+            sort_order,
+            delimiter,
+            metric_first_label,
+        )
+        field_check_expr.update(field_check_expr_s)
+        field_check_details.update(field_check_details_s)
+
+        generic_field_check_loop(t, field_check_expr, field_check_details, verbose)
```

Comment:
PART OF LARGE SUGGESTION
```suggestion

        generic_field_check_loop(t, field_check_expr, verbose)
```